test/loguniform.jl

```diff
@@ -0,0 +1,50 @@
+module TestLogUniform
+using Test
+using Distributions
+import Random
+
+@testset "LogUniform" begin
+    rng = Random.MersenneTwister(0)
+    d = LogUniform(1,10)
+    @test 1 <= rand(rng, d) <= 10
+    @test typeof(rand(rng, d) ) == eltype(d)
```

Comment:
A bit simpler:
```suggestion
    @test rand(rng, d) isa eltype(d)
```

Reply:
Mhh `eltype(d) = Any` would pass the test you suggest. This is undesirable I think?

Reply:
Ah, I see your point. If your interested in `eltype(d)` and not that `rand(rng, d)` is of `eltype(d)`, then I would suggest checking it directly:
```suggestion
    @test eltype(d) === Float64
    @test rand(rng, d) isa eltype(d)
```
Generally, it is preferable to check equality of types and singletons with `===`.

Reply:
I am a bit confused now. What information does the eltype of a distribution encode? Is `eltype(d) === typeof(rand(rng, d))` expected to hold in general? How about `typeof(pdf(d, 0)) === eltype(d)`?

Reply:
> What information does the eltype of a distribution encode? Is eltype(d) === typeof(rand(rng, d)) expected to hold in general?

According to the docstring of `eltype` it should return the element type of a sample. But there's a lot of issues and inconsistencies with how it is implemented currently. I suggest we don't focus on this too much here (I hope to improve the situation soon) but just use the same approach as in most other distributions currently.

> How about typeof(pdf(d, 0)) === eltype(d)?

No, the return type of `pdf`, `cdf` etc. is not enforced and is just whatever the function returns. One main motivation is that it also depends on the type of `x` (the point where these functions are evaluated).